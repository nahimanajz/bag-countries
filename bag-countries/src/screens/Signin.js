import { Mail } from "../assets/Mail";

const { Lock } = require("../assets/Lock");

export default function Signin() {

    return(
            <form action="/" method="POST" className="modal">
                <div style={form}>
                <div className="grid-col-2">
                    <span><Mail /></span>
                        <div>
                            <input type="email" placeholder="Email" name="email" required className="input"/>
                            {/* {/*1===1&&<small>LmORARfEVT1iyjwN*</small> */} 
                         </div>
                </div>
                <div className="grid-col-2">                       
                    <span><Lock /></span>
                    <div>
                        <input type="password" placeholder="Email" name="email" required/>
                        {/*1===1&&<small>LmORARfEVT1iyjwN*</small> */}
                    </div>
                </div>
                <div>
                    <input className="btn" type="submit" value="Signin" />
                </div>
                </div>
            </form>

            
    )
}
const form = {display: 'grid', gap: 30, gridTemplateRows:'repeat(3, 1fr)'} 