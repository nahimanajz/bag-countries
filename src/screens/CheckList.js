import axios from "axios";
import { useState } from "react";
import { toast, ToastContainer } from "react-toastify";
import { CountryFilter } from "../components/CountryFilter";
import { ListableCountry } from "../components/ListableCountry";
import { toLowerCase, BACKEND_API_ROUTE } from "../util";

/**
 * Loop countries on user UI screens
 * @param {Array} countries fetched countries from App component to pass to Listable item
 * @param {Object} userInfo signed user
 @return {Component, Component} Listable Country filter
 */

export function CheckList({ countries, userInfo, ...props }) {
  const [myList, setMyList] = useState();
  const [regionCountries, setRegionCountries] = useState();

  const addCountryToList = (e) => {
    const country = countries.filter(({ name }) =>
      toLowerCase(name).includes(toLowerCase(e.target.value))
    );
    setMyList(country); 
    setRegionCountries();
  };
  const addCountryByRegion = async (e) => {
    const country = countries.filter(
      ({ region }) => toLowerCase(region) === toLowerCase(e.target.value)
    );
    setRegionCountries(country);
    setMyList(country);
  };
  //Todo: addTo list in backend
  const handleCheckList = async (name) => {
   
    const { data } = await axios.post(`${BACKEND_API_ROUTE}/add/country`, {
      name,
      user: userInfo._id,
    });
    if (data) {
      toast(data.message);
    } else {
      toast("Something is wrong");
    }
  };

  return (
    <>
      <ToastContainer />
      <CountryFilter
        addCountryToList={addCountryToList}
        addCountryByRegion={addCountryByRegion}
      />
      <ListableCountry
        goToDetail={props.goToDetail}
        add={handleCheckList}
        countries={myList ? myList : regionCountries && regionCountries}
      />
    </>
  );
}
