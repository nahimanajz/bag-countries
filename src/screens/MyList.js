import { useEffect, useState } from "react";
import { CountryFilter } from "../components/CountryFilter";
import { ListableCountry } from "../components/ListableCountry";
import { toLowerCase } from "../util";
/**
 * - Countries to Visit  
 * @param {Function} changeScreenTitle to change screen to appropriat visited screen
 * @param {Function} updateCountry  to updated selected country to visited list
 * @param {Function} deleteCountry  to remove country from visit list
 * @param {Array} countries fetched countries from App component to pass to Listable item
 * @param {*} JSX tags to append on my list screen
 */

export function MyList({
  countries,
  children,
  changeScreenTitle,
  updateCountry,
  deleteCountry,
  ...props
}) {
  const [myList, setMyList] = useState();
  const [regionCountries, setRegionCountries] = useState();
  const [countryItem, setCountryItem] = useState();
  
  const addCountryToList = (e) => {
    const country = countries.filter(({ name }) =>
      toLowerCase(name).includes(toLowerCase(e.target.value))
    );
    setMyList(country);
    setRegionCountries();
  };
  const addCountryByRegion = async (e) => {
    const country = countries.filter(
      ({ region }) => toLowerCase(region) === toLowerCase(e.target.value)
    );
    setRegionCountries(country);
    setMyList(country);
  };
  useEffect(() => {
    setCountryItem(myList ? myList : regionCountries && regionCountries);
  }, [myList, regionCountries]);

  return (
    <div className="flex-col">
      <CountryFilter
        addCountryToList={addCountryToList}
        addCountryByRegion={addCountryByRegion}
      />
      <ListableCountry
        countries={countryItem}
        goToDetail={props.goToDetail}
        updateCountry={updateCountry}
        deleteCountry={deleteCountry}
      />
    </div>
  );
}
