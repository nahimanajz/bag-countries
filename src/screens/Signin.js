import axios from 'axios';
import { useCallback, useState } from 'react'
import {useNavigate } from "react-router-dom";
import { Mail } from "../assets/Mail";
import { BACKEND_API_ROUTE } from "../util";
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

export default function Signin(props) {

   const[state, setState] = useState({email:'', password:''})
   const handleChange = useCallback(
    (e) => {
      setState({...state, [e.target.name]: e.target.value});
    },
    [state],
  );
   


   const navigate = useNavigate();
    const handleSignin =  async()=>{     
        const{email, password} = state;  
        const{ data}  = await axios.post(`${BACKEND_API_ROUTE}/signin`,{email, password})
        if(!data.message){   
            navigate('/my-list',{loggedUser:data})
            props.showDashboard(true)
            props.userInfo(data)
           
        }
        toast(data.message);
        }
       
    return(
        <>
            <ToastContainer />
            <form method="post">          
		      <ul className="modal">
                 <li>
                    <label>Email</label>        
                       <input type="email" onChange={handleChange} placeholder="Email" name="email" required />
                  </li>  
                  <li>
                        <label>Password</label>        
                        <input type="password" onChange={handleChange} placeholder="Password" name="password" required/>
                  </li>
                  <li>                     
                     <input  type="button"
                        onClick={handleSignin} 
                        className="btn" 
                      value="Signin" style={{height: 48}} 
                     /> 
                  </li>      
              </ul>
            </form>
        </>
            
    )
}