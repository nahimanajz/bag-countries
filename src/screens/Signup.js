import { useCallback, useState } from 'react'
import axios from 'axios'
import { BACKEND_API_ROUTE } from '../util'
import { toast, ToastContainer } from 'react-toastify';

/**
 * - User signin 
 * @param {Function} props.showDashboard to show main dashboard later on sinup
 * @param {Function} props.userInfo  setUser info so that can view relavant info on dashboard
 */

export function Signup({ showDashboard, userInfo}){
   const[state, setState] = useState({name:'', email:'', password:'',phone:'',dob:'',country:''})
   const handleChange = useCallback(
    (e) => {
      setState({...state, [e.target.name]: e.target.value});
    },
    [state],
  );
   const handleSignup= async ()=>{
   const {name, email, password,phone,dob,country} = state;
   const{ data}  = await axios.post(`${BACKEND_API_ROUTE}/signup`,{name, email, password,phone,dob,country})
  

    if(data.error){ 
       toast('Please correct your data');

    }else{
      toast('Well done!!')
      userInfo(data.newUser);
      showDashboard(true)
      
    }
    
   }

    return(
         <>
            <ToastContainer />
            <form method="post">
		      <ul className="modal">
               <li>
                  <label>Full Names</label>        
                  <input type="text" onChange={handleChange} placeholder="Full Names" name="name" required />
                      
                  </li>
                  <li>
                        <label>Email</label>        
                        <input type="email" onChange={handleChange} placeholder="Email" name="email" required />
                  </li>  
                  <li>
                        <label>Password</label>        
                        <input type="password" onChange={handleChange} placeholder="Password" name="password" required/>
                  </li> 
                  <li>
                        <label>Phone</label>        
                        <input type="tel" onChange={handleChange} placeholder="Phone" name="phone" required/>
                  </li> 
                  <li>
                     <label>DOB</label>
                     <input type="date" onChange={handleChange} placeholder="Date Of Birth" name="dob" required/>
                  </li> 
                 
                  <li>
                     <label>Country</label>
                     <input type="text" onChange={handleChange} placeholder="Country" name="country" required/>
                     
                  </li>
                  <li>                     
                     <input  type="button"
                        onClick={handleSignup} 
                        className="btn" 
                      value="Signup" style={{height: 48}} 
                     disabled={state.length <6?true: false}/> 
                  </li>    
               
               </ul>
                </form>
         </>
    )
}