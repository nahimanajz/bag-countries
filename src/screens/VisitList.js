
import { ListableCountry } from "../components/ListableCountry";

export default function VisitList({countries, goToDetail, updateCountry, deleteCountry}){

   return (
       <>
      <ListableCountry
        countries={countries}
        goToDetail={goToDetail}
        updateCountry={updateCountry}
        deleteCountry={deleteCountry}
        
      />
      </>
   )
}