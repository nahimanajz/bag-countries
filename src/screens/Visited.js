import { ListableCountry } from "../components/ListableCountry";

export default function Visited({countries, goToDetail, updateCountry, deleteCountry}){

    return(
            <ListableCountry
            countries={countries}
            goToDetail={goToDetail}
            updateCountry={updateCountry}
            deleteCountry={deleteCountry}
            checked={true}
        />
        )
}