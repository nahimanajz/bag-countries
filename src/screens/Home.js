import { useState } from "react";
import Signin  from './Signin';
import { Signup } from "./Signup";

/**
 * Toggle Login and Signup screens
 * @param {Function} props.setShowDashoard to show main dashboard later on logging in or successfull signup
 */

export default function Home(props) {
    const[showLogin, setShowLogin] = useState(false);
    const switchScreens = ()=> setShowLogin(!showLogin);

    return(<>
    
        <div className="cover">
            <div className="model-header">
                <div className={showLogin?'inactive':'active'} onClick={switchScreens}>Signup</div>
                <div className={showLogin?'active':'inactive'} onClick={switchScreens}>Signin</div>
            </div>
            {showLogin? <Signin showDashboard={props.setShowDashboard} userInfo={props.userInfo}/>:<Signup showDashboard={props.setShowDashboard} userInfo={props.userInfo}/> }
             
        </div>
        
        </>
    )
}
