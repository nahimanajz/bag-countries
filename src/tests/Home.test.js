import React from 'react';
import { shallow } from 'enzyme';
import Home from '../screens/Home';
import { Signup } from '../screens/Signup';
import Signin from '../screens/Signin';

describe("render dashboard screens", () => {
    it("should render home component without error", () => {
      const wrapper = shallow(<Home />);
      expect(wrapper.getElements()).toMatchSnapshot();
    });
    it("should render Signup component without error", () => {
      const wrapper = shallow(<Signup />);
      expect(wrapper.getElements()).toMatchSnapshot();
    });
    it("should render Signin component without error", () => {
      const wrapper = shallow(<Signin />);
      expect(wrapper.getElements()).toMatchSnapshot();
    });
    
  });