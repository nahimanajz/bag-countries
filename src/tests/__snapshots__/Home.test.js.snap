// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dashboard screens should render Signup component without error 1`] = `
Array [
  <React.Fragment>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      theme="light"
      transition={[Function]}
    />
    <form
      method="post"
    >
      <ul
        className="modal"
      >
        <li>
          <label>
            Full Names
          </label>
          <input
            name="name"
            onChange={[Function]}
            placeholder="Full Names"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label>
            Email
          </label>
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
          />
        </li>
        <li>
          <label>
            Password
          </label>
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
        </li>
        <li>
          <label>
            Phone
          </label>
          <input
            name="phone"
            onChange={[Function]}
            placeholder="Phone"
            required={true}
            type="tel"
          />
        </li>
        <li>
          <label>
            DOB
          </label>
          <input
            name="dob"
            onChange={[Function]}
            placeholder="Date Of Birth"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label>
            Country
          </label>
          <input
            name="country"
            onChange={[Function]}
            placeholder="Country"
            required={true}
            type="text"
          />
        </li>
        <li>
          <input
            className="btn"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "height": 48,
              }
            }
            type="button"
            value="Signup"
          />
        </li>
      </ul>
    </form>
  </React.Fragment>,
]
`;

exports[`render dashboard screens should render home component without error 1`] = `
Array [
  <React.Fragment>
    <div
      className="cover"
    >
      <div
        className="model-header"
      >
        <div
          className="active"
          onClick={[Function]}
        >
          Signup
        </div>
        <div
          className="inactive"
          onClick={[Function]}
        >
          Signin
        </div>
      </div>
      <Signup />
    </div>
  </React.Fragment>,
]
`;

exports[`render dashboard screens should render visit list  component without error 1`] = `
Array [
  <React.Fragment>
    <div
      className="cover"
    >
      <div
        className="model-header"
      >
        <div
          className="active"
          onClick={[Function]}
        >
          Signup
        </div>
        <div
          className="inactive"
          onClick={[Function]}
        >
          Signin
        </div>
      </div>
      <Signup />
    </div>
  </React.Fragment>,
]
`;
